src/Features/LanguageServer/Protocol/Handler/Configuration/DidChangeConfigurationNotificationHandler.cs

```diff
@@ -0,0 +1,212 @@
+                if (option is IPerLanguageValuedOption && languageName != null)
+                {
+                    _globalOptionService.SetGlobalOption(new OptionKey2(option, language: languageName), result);
+                }
+                else
+                {
+                    _globalOptionService.SetGlobalOption(new OptionKey2(option, language: null), result);
+                }
+            }
+        }
+
+        private async Task<ImmutableArray<string>> GetConfigurationsAsync(CancellationToken cancellationToken)
+        {
+            try
+            {
+                var configurationParams = new ConfigurationParams() { Items = _configurationItems.AsArray() };
+                var options = await _clientLanguageServerManager.SendRequestAsync<ConfigurationParams, JArray>(
+                    Methods.WorkspaceConfigurationName, configurationParams, cancellationToken).ConfigureAwait(false);
+
+                // Failed to get result from client.
+                Contract.ThrowIfNull(options);
+                return options.SelectAsArray(token => token.ToString());
+            }
+            catch (Exception e)
+            {
+                _lspLogger.LogException(e, $"Exception occurs when make {Methods.WorkspaceConfigurationName}.");
+            }
+
+            return ImmutableArray<string>.Empty;
+        }
+
+        private static ImmutableArray<ConfigurationItem> GenerateGlobalConfigurationItems()
+        {
+            using var _ = ArrayBuilder<ConfigurationItem>.GetInstance(out var builder);
+            foreach (var option in SupportedOptions)
+            {
+                var fullOptionName = GenerateFullNameForOption(option);
+                if (option is IPerLanguageValuedOption)
+                {
+                    foreach (var language in SupportedLanguages)
+                    {
+                        builder.Add(new ConfigurationItem()
+                        {
+                            Section = string.Concat(s_languageNameToPrefix[language], '.', fullOptionName),
+                        });
+                    }
+                }
+                else
+                {
+                    builder.Add(new ConfigurationItem()
+                    {
+                        Section = fullOptionName,
+                    });
+                }
+            }
+
+            return builder.ToImmutable();
+        }
+
+        private static string GenerateFullNameForOption(IOption2 option)
+        {
+            var optionGroupName = GenerateOptionGroupName(option);
+            // All options send to the client should have group name and config name.
+            RoslynDebug.Assert(!string.IsNullOrEmpty(optionGroupName));
+            RoslynDebug.Assert(!string.IsNullOrEmpty(option.Definition.ConfigName));
+            return string.Concat(optionGroupName, '.', option.Definition.ConfigName);
+        }
+
+        private static string GenerateOptionGroupName(IOption2 option)
```

Comment:
could you put a comment in the code with an example with an option and what the generation outputs?  Just to make it easier at a glance to seewhats going on